test/jdk/javax/swing/text/DefaultEditorKit/4278839/bug4278839.java

```diff
@@ -61,24 +61,53 @@ public static void main(String[] args) throws Exception {
 
             clickMouse();
             robo.waitForIdle();
+            robo.delay(250);
 
             area.setCaretPosition(0);
             robo.waitForIdle();
+            robo.delay(250);
 
             passed &= moveCaret(true) == 1;
+            System.out.println("passed1 moveCaret1 " + passed);
```

Comment:
If you are trying to get more information here why not logging the actual position of the caret after the move? Either inside the moveCaret() method or just add a new variable like
`int caret;`
And then every time you do the test you assign it.
`passed &= (caret = moveCaret(true)) == 1;`
and then you can just output the new caret position after each iteration like "Expected position 1 actual position " + caret.

Reply:
good point..logged..